core/lifecycle.go

```diff
@@ -14,13 +14,34 @@
 
 package core
 
-// Lifecycle is a lifecycle service for Hazelcast clients.
-type Lifecycle interface {
-	// AddListener adds a listener object to listen for lifecycle events.
-	// AddListener returns the registrationID.
-	AddListener(listener interface{}) string
-
-	// RemoveListener removes lifecycle listener with the given registrationID.
-	// RemoveListener returns true if the listener is removed successfully, false otherwise.
-	RemoveListener(registrationID string) bool
+const (
+	// LifecycleStateStarting is fired when the client is starting.
+	LifecycleStateStarting = "STARTING"
+
+	// LifecycleStateStarted is fired when the client start is completed.
+	LifecycleStateStarted = "STARTED"
+
+	// LifecycleStateConnected is fired when the client is connected to a member.
+	LifecycleStateConnected = "CONNECTED"
+
+	// LifecycleStateDisconnected is fired when the client disconnected from a member.
+	LifecycleStateDisconnected = "DISCONNECTED"
+
+	// LifecycleStateShuttingDown is fired when the client is shutting down.
+	LifecycleStateShuttingDown = "SHUTTING_DOWN"
+
+	// LifecycleStateShutdown is fired when the client shutdown is completed.
+	LifecycleStateShutdown = "SHUTDOWN"
```

Comment:
Is this camel case because of go convention ? 
If there is no convention from go, I think  as capital letters divided by underscore is more appropriate. 
But if this is the way from go world, following go convention makes more sense.

Reply:
It is the convention in Go.